src/LogFileReader.cpp

```diff
@@ -47,12 +51,25 @@ bool LogFileReader::readAll(const QString& fileName)
 // File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
-    // TODO implement this first
+    //example input= 07:01:07.024, 3.538654, 25.523571
+    bool convertedOkay;
+    QStringList lineSplit = line.split(",");
+    if(lineSplit.length()!= 3)
+        {return false;}
+
+    QTime time = QTime::fromString(lineSplit[0],TIME_FORMAT);
+    if(!time.isValid())
+        {return false;}
+
+    double voltage = lineSplit[1].toDouble(&convertedOkay);
+    if(!convertedOkay||voltage<0)
+        {return false;}
+
+    double current = lineSplit[2].toDouble(&convertedOkay);
+    if(!convertedOkay)
+        {return false;}
```

Comment:
These `if` statements I would give two ways of writing them:
```c++
if(!convertedOkay) { return false; }
```
Note the spaces around the `{` brackets.

The other way that we see more regularly is:
```c++
if(!convertedOkay)
{
    return false;
}
```